templates/react-native/src/services/template.ts.twig

```diff
@@ -0,0 +1,182 @@
+
+{% endfor %}
+{% endfor %}
+{% endif %}
+
+        for (const [key, value] of Object.entries(Service.flatten(payload))) {
+            uri.searchParams.append(key, value);
+        }
+{% endif %}
+{% if method.type == 'webAuth' %}
+        return uri;
+{% elseif method.type == 'location' %}
+        return uri;
+{% else %}
+{% if 'multipart/form-data' in method.consumes %}
+{% for parameter in method.parameters.all %}
+{% if parameter.type == 'file' %}
+
+        const size = {{ parameter.name | caseCamel | escapeKeyword }}.size;
+
+        if (size <= Service.CHUNK_SIZE) {
+            return await this.client.call('{{ method.method | caseLower }}', uri, {
+{% for parameter in method.parameters.header %}
+                '{{ parameter.name | caseCamel | escapeKeyword }}': this.client.${{ parameter.name | caseCamel | escapeKeyword }},
+{% endfor %}
+{% for key, header in method.headers %}
+                '{{ key }}': '{{ header }}',
+{% endfor %}
+            }, payload);
+        }
+
+        const apiHeaders: { [header: string]: string } = {
+{% for parameter in method.parameters.header %}
+            '{{ parameter.name | caseCamel | escapeKeyword }}': this.client.${{ parameter.name | caseCamel | escapeKeyword }},
+{% endfor %}
+{% for key, header in method.headers %}
+            '{{ key }}': '{{ header }}',
+{% endfor %}
+        }
+
+        let offset = 0;
+        let response = undefined;
+{% for parameter in method.parameters.all %}
+{% if parameter.isUploadID %}
+        if({{ parameter.name | caseCamel | escapeKeyword }} != 'unique()') {
+            try {
+                response = await this.client.call('GET', new URL(this.client.config.endpoint + apiPath + '/' + {{ parameter.name }}), apiHeaders);
+                offset = response.chunksUploaded * Service.CHUNK_SIZE;
+            } catch(e) {
+            }
+        }
+{% endif %}
+{% endfor %}
+
+        let timestamp = new Date().getTime();
+        while (offset < size) {
+            let end = Math.min(offset + Service.CHUNK_SIZE - 1, size - 1);
+
+            apiHeaders['content-range'] = 'bytes ' + offset + '-' + end + '/' + size;
+            if (response && response.$id) {
+                apiHeaders['x-{{spec.title | caseLower }}-id'] = response.$id;
+            }
+
+            let chunk = await FileSystem.readAsStringAsync({{ parameter.name | caseCamel | escapeKeyword }}.uri, {
+                encoding: FileSystem.EncodingType.Base64,
+                position: offset,
+                length: Service.CHUNK_SIZE
+            });
+            var path = `data:${{'{'}}{{ parameter.name | caseCamel | escapeKeyword }}.type{{'}'}};base64,${{'{'}}chunk{{'}'}}`;
+            if (Platform.OS.toLowerCase() === 'android') {
+                path = FileSystem.cacheDirectory + '/tmp_chunk_' + timestamp;
+                await FileSystem.writeAsStringAsync(path, chunk, {encoding: FileSystem.EncodingType.Base64});
+            }
```

Comment:
Is there no other way to do this? It will be slow, especially on Android

Reply:
If there is I've not found one yet